router/src/infer/tool_grammar.rs

```diff
@@ -20,19 +19,19 @@ impl ToolGrammar {
 
     pub fn apply(
         tools: Vec<Tool>,
-        tool_choice: ChatCompletionToolChoiceOption,
+        tool_choice: ToolChoice,
     ) -> Result<(Vec<Tool>, Option<JsonSchemaTool>), InferError> {
-        // if no tools are provided, we return None
+        // if no tools are provided, we return None and an empty vec
         if tools.is_empty() {
-            return Ok((tools, None));
+            return Ok((Vec::with_capacity(0), None));
```

Comment:
`vec![]` ?

Reply:
updated in latest commit

Reply:
I don't see it.

Reply:
`let tools_to_use = match tool_choice` was updated to return a `vec![]` rather than a `Vec::with_capacity(0)`, and the conditional was moved below `tools_to_use` to return none in both cases (no tools supplied, or if NoTools specified) https://github.com/huggingface/text-generation-inference/blob/16e4c38898ed36ee44237314ce823bf3d81ea90c/router/src/infer/tool_grammar.rs#L55